src/Analysis/Ast/Impl/Types/Definitions/IPythonFunctionOverload.cs

```diff
@@ -50,17 +50,17 @@ public interface IPythonFunctionOverload {
         /// Determines return value type given arguments for the particular call.
         /// For annotated or stubbed functions the annotation type is always returned.
         /// </summary>
-        IMember GetReturnValue(LocationInfo callLocation, IArgumentSet args);
+        /// <param name="args">Call arguments or type arguments.</param>
+        /// <param name="self">Invoking class instance. In case of generics it is instance of the specific type
+        /// as opposed to declaring type which is the generic template class.</param>
+        /// <param name="callLocation">Call expression location, if any.</param>
+        IMember Call(IArgumentSet args, IPythonType self, LocationInfo callLocation = null);
 
         /// <summary>
         /// Return value documentation.
         /// </summary>
-        string ReturnDocumentation { get; }
-
-        /// <summary>
-        /// Function definition is decorated with @overload.
-        /// </summary>
-        bool IsOverload { get; }
+        /// <param name="self">If function is in generic class it may need the specific type to determine the return type.</param>
+        string GetReturnDocumentation(IPythonType self);
```

Comment:
Should this be `IPythonType self = null` to get rid of all of the calls that look like `GetReturnDocumentation(null)`, and to better indicate that `self` is allowed to be null? I can see other functions you've modified that do `self = null`, but wasn't sure if you were skipping this specific one to make it explicit that there is no `self`.

Reply:
Sure why not. I made it without `= null` so I could find and fix relevant cases.

Reply:
That's what I had guessed; if keeping `(null)` prevents future errors, like forgetting to specify `self`, then I wouldn't remove it.